packages/app-desktop/gui/NoteEditor/utils/resourceHandling.ts

```diff
@@ -125,3 +126,40 @@ export async function handlePasteEvent(event: any) {
 	}
 	return output;
 }
+
+export async function processPastedHtml(html: string) {
+	const allImageUrls: string[] = [];
+	const mappedResources: Record<string, string> = {};
+
+	htmlUtils.replaceImageUrls(html, (src: string) => {
+		allImageUrls.push(src);
+	});
+
+	for (const imageSrc of allImageUrls) {
+		if (!mappedResources[imageSrc]) {
+			try {
+				if (imageSrc.startsWith('file')) {
+					const imageFilePath = imageSrc.substr(7);
+
+					if (imageFilePath.startsWith(Setting.value('resourceDir'))) {
+						mappedResources[imageSrc] = imageSrc;
+					} else {
+						const createdResource = await shim.createResourceFromPath(imageFilePath);
+						mappedResources[imageSrc] = `file://${Resource.fullPath(createdResource)}`;
+					}
+				} else {
+					const filePath = `${Setting.value('tempDir')}/${md5(Date.now())}`;
+					await shim.fetchBlob(imageSrc, { path: filePath });
+					const createdResource = await shim.createResourceFromPath(filePath);
+					mappedResources[imageSrc] = `file://${Resource.fullPath(createdResource)}`;
```

Comment:
temp file needs to be deleted after the resource has been created

Reply:
okay, done.